reactioncommerce/reaction-next-starterkit
# Obey url rewrites in browser-side routing

Resolves #452 
Impact: **minor**
Type: **feature**

## Issue

Rewrite rules by nature mask the actual URL the underlying application uses for routing. This is problematic when dealing with single-page app routing as it has no idea that route exists, resulting in 404's.

## Solution

Update starter-kit to accept a `Request-Path` from the proxy handling the rewrite rules in order to make a new route entry in the browser-side app.

## Breaking changes

none.

## Testing
1. With the reaction platform, and steps outlined in other repos
1. Start reaction platform
1. In the Redirect URLs admin panel, add a new rewrite rule for a tag page
1. For example from: `/example/route` to: `/tag/shop`
1. Navigate to `localhost:9090/example/route`
1. See the tag page you expect
1. Adjust the page limit/price filters
1. See that the URL in the address bar remains `/example/route`
